utils/s2n_compiler.h

```diff
@@ -22,3 +22,11 @@
 #define S2N_GCC_VERSION_AT_LEAST(major, minor, patch_level) \
     ((S2N_GCC_VERSION) >= ((major) * 10000 + (minor) * 100 + (patch_level)))
 
+#if defined (__clang__)
+    #define S2N_GCC_VERSION_AT_LEAST_OR_CLANG(major, minor, patch_level) \
```

Comment:
minor: I think flipping it would be more readable `S2N_CLANG_OR_GCC_VERSION_AT_LEAST`

Reply:
I was trying to keep the naming similar to the above naming `S2N_GCC_VERSION_AT_LEAST` since I think that helps with discoverability(IDE) and readability.